cmd/process-agent/main_common.go

```diff
@@ -74,8 +74,17 @@ var (
 )
 
 func getSettingsClient() (settings.Client, error) {
-	// Set up the config in case the cmd_port was specified
-	_, err := config.NewAgentConfig(loggerName, opts.configPath, opts.sysProbeConfigPath)
+	// Set up the config so we can get the port later
+	// We set this up differently from the main process-agent because this way is quieter
+	cfg := config.NewDefaultAgentConfig(false)
+	if opts.configPath != "" {
+		ddconfig.Datadog.SetConfigFile(opts.configPath)
```

Comment:
Will this still work if the user only informs the path of the `datadog.yaml` file, like `./process-agent --config /etc/datadog/` config get log_level? 

Looking at the `loadConfigIfExists` function used by `NewAgentConfig`, we're also calling `AddConfigPath` if the file was not specified in the path.

https://github.com/DataDog/datadog-agent/blob/4452508ba1985be3a01320ded1faccb11dc33991/pkg/process/config/config.go#L258-L274

Is there any side effect if we export this `loadConfigIfExists` function and use it here?

Reply:
Per our discussion in slack, we decided the best thing to do is to export config.loadConfigIfExists and use that, so that you can specify a directory for the config instead of the file path directly.